assets/js/dataclip-viewer/component.tsx

```diff
@@ -37,35 +37,96 @@ async function fetchDataclipContent(dataclipId: string) {
 
 const DataclipViewer = ({ dataclipId }: { dataclipId: string }) => {
   const [content, setContent] = useState<string>('');
+  const ref = useRef<HTMLDivElement>(null);
   const monacoRef = useRef<Monaco | null>(null);
+  const editorRef = useRef<any>(null);
+  const resizeObserver = useRef<any>(null);
 
   const beforeMount = (monaco: Monaco) => {
     monacoRef.current = monaco;
     setTheme(monaco);
   };
 
+  const handleEditorDidMount = useCallback((editor: any) => {
+    editorRef.current = editor;
+  }, []);
+
+  useEffect(() => {
+    if (editorRef.current && ref.current && !resizeObserver.current) {
+      console.debug('Setting up editor resize observer');
+      let timeoutId: NodeJS.Timeout;
+      const handleResize = () => {
+        function doResize() {
+          let editor = editorRef.current;
+          editor.layout({ width: 0, height: 0 });
+          editor.layout();
+        }
+        if (timeoutId) {
+          clearTimeout(timeoutId);
+          doResize();
+        } else {
+          timeoutId = setTimeout(() => {
+            let editor = editorRef.current;
+            editor.layout({ width: 0, height: 0 });
+            editor.layout();
+          }, 100);
+        }
+      };
+
+      resizeObserver.current = new ResizeObserver(
```

Comment:
HEy @stuartc as we just discussed:

Can we bookmark this implementation, merge my branch, and if we think my branch is suboptimal (and it likely is!) we can paste THIS implementation into the new Monaco wrapper.

That simplifies this PR, unblocks Frank, and lets us demote the severity of this to "optimisation"

I do want to keep a copy of this code though because I like the "don't resize hidden editors thing! (what happens if the editor is hidden, the window is resize, and the editor is revealed? We may have to kick it on tab select or something)